src/Core/Services/ExecutionHelper.cs

```diff
@@ -319,8 +321,8 @@ private static void SetContextResult(IMiddlewareContext context, JsonDocument? r
                 SupportedHotChocolateTypes.INT_TYPE => ((IntValueNode)value).ToInt32(),
                 SupportedHotChocolateTypes.LONG_TYPE => ((IntValueNode)value).ToInt64(),
                 SupportedHotChocolateTypes.SINGLE_TYPE => ((FloatValueNode)value).ToSingle(),
-                SupportedHotChocolateTypes.FLOAT_TYPE => ((FloatValueNode)value).ToDouble(),
-                SupportedHotChocolateTypes.DECIMAL_TYPE => ((FloatValueNode)value).ToDecimal(),
+                SupportedHotChocolateTypes.FLOAT_TYPE => value is IntValueNode intValueNode ? intValueNode.ToDouble() : ((FloatValueNode)value).ToDouble(),
+                SupportedHotChocolateTypes.DECIMAL_TYPE => value is IntValueNode intValueNode ? intValueNode.ToDecimal() : ((FloatValueNode)value).ToDecimal(),
                 SupportedHotChocolateTypes.UUID_TYPE => Guid.TryParse(value.Value!.ToString(), out Guid guidValue) ? guidValue : value.Value,
```

Comment:
Can we do `Float.Parse(value.Value)` so that all kinds of valid float values get parsed?

Reply:
Thanks, Ani, I tried the suggested change but the .Net float type has different range than Hotchocolate floatNodeType, which is breaking existing tests.
The float type in .NetC# has a maximum value of approximately 3.4E38. Any value larger than this will exceed the range of float, and when we try to parse such a value, C# will return Infinity, and we have a test for 2E150 which started failing.
float.Parse() for any larger value gives `Infinity`
![image](https://github.com/Azure/data-api-builder/assets/102276754/d6decab3-7757-48c1-89b4-fdfbf928f55c)

Reply:
Also noticed other tests failing like below:
![image](https://github.com/Azure/data-api-builder/assets/102276754/803be145-b2dd-4456-9417-9fb4a5de72b3)